sys-cluster/hpx: Bump version

shifty91 opened the pull request:
Add the current HPX version: http://stellar-group.org/2020/02/hpx-1-4-1-released/. 
The documentation generation requires an update to date dev-python/breathe version.

Package-Manager: Portage-2.3.89, Repoman-2.3.20
Signed-off-by: Kurt Kanzenbach <kurt@kmk-computers.de>

juippis commented on dev-python/breathe/breathe-4.14.1.ebuild:
```diff
@@ -0,0 +1,29 @@
+
+DESCRIPTION="Sphinx Doxygen renderer"
+HOMEPAGE="https://breathe.readthedocs.io/en/latest/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
```
https://github.com/michaeljones/breathe looks like at least Github has tests listed, although I didnt download a tarball to confirm they are packaged. Could you check?

shifty91 replied:
The tests are not packaged.

juippis commented on sys-cluster/hpx/hpx-1.4.1.ebuild:
```diff
@@ -0,0 +1,99 @@
+"
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-libs/boost
```
Are you sure this isn't linked? Boost is usually quite ABI sensitive, please check.

shifty91 replied:
Boost is linked e.g. into the libhpx.so. The original ebuild used `dev-libs/boost:=`. Is that the correct way?

shifty91 replied:
Hm, now I get it (https://wiki.gentoo.org/wiki/Sub-slots_and_Slot-Operators). Should be fixed now. Thanks!

juippis commented on sys-cluster/hpx/hpx-1.4.1.ebuild:
```diff
@@ -0,0 +1,99 @@
+	${PYTHON_DEPS}
+	dev-libs/boost
+	sys-apps/hwloc
+	sys-libs/libunwind
```
Is this really an unconditional dependency on every architechture? Seems weird.

shifty91 replied:
No, it's not. The documentation for 1.4.1 states that libunwind is needed whenever google-perftools is used on x86_64 (https://stellar-group.github.io/hpx-docs/latest/html/manual/building_hpx.html). The ebuild google-perftools already includes the dependency to libunwind. So, that one here can be removed.

shifty91 commented:
Why does it fail? Out of memory, maybe? hpx is c++ heavy. It works for me with twice as much RAM in G as cores (16 cores -> 32 GiB RAM). But, what can I do about it?

juippis commented:
Oh, wow, it does indeed. Never thought I'd see that happen.

You may want to utilize *check-reqs.eclass* because it's better to stop the user at `pkg_setup` instead of leaving them with filled tmpfs for example. I don't have a straight example how to use it with scaling values, but I imagine you could do it in steps. Something like 
```
if [[ ${MAKEOPTS} -ge j4 ]]; then
	CHECKREQS_MEMORY=8G

elif [[ ${MAKEOPTS} -ge j8 ]]; then
	CHECKREQS_MEMORY=16G

elif [[ ${MAKEOPTS} -ge j12 ]]; then
	CHECKREQS_MEMORY=24G

elif [[ ${MAKEOPTS} -ge j16 ]]; then
	CHECKREQS_MEMORY=32G

else
	CHECKREQS_MEMORY=2G
```

probably there are better ways, but you get the idea I'm after. Maybe you can even just take the number from MAKEOPTS and multiply it by 2 and add that to `CHECKREQS_MEMORY`.

shifty91 commented:
1G RAM seems to be enough per job. Added that to the ebuild. Also switched to `cmake_src_test` and called python setup unconditionally, which fixes builds without `doc` and `test`.
But, I have no idea why the warning about the boost root is issued. Boost root is actually considered and this only happens when the tests are executed...

shifty91 commented:
@juippis Found the boost cmake problem. Should be fixed now. Thanks for the review.

shifty91 commented:
The tests shouldn't be executed in parallel. Anyway, the ebuild should take care of that. Yes, that needs further research. Thanks!
